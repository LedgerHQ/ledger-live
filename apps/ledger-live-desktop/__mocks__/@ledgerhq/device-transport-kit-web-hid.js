class WebHidSendReportError extends Error {}

module.exports = {
  WebHidSendReportError,
};
