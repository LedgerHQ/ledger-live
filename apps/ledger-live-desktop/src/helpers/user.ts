import { v4 as uuid } from "uuid";
import { setKey, getKey } from "~/renderer/storage";

// a user is an anonymous way to identify a same instance of the app

// only used by analytics. DEPRECATED (will will later switch to localStorage)
export default async () => {
  let user = await getKey("app", "user");
  if (!user) {
    user = {
      id: uuid(),
    };
    setKey("app", "user", user);
  }
  return user;
};
export const getUserId = () => {
  if (typeof window === "object") {
    const { localStorage } = window;
    let userId = localStorage.getItem("userId");
    if (!userId) {
      userId = uuid();
      localStorage.setItem("userId", userId);
      return userId;
    }
    return localStorage.getItem("userId");
  }
  throw new Error("user is only to be called from renderer");
};
