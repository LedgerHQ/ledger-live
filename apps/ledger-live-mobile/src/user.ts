import { v4 as uuid } from "uuid";
import { getUser, setUser, updateUser as _updateUser } from "./db";
// Nb a user is an anonymous way to identify a same instance of the app
let user;

async function updateUser() {
  user = {
    id: uuid(),
  };
  await _updateUser(user);
  return {
    user,
    created: false,
  };
}

export default async () => {
  if (!user) {
    user = await getUser();
  }

  if (!user) {
    user = {
      id: uuid(),
    };
    await setUser(user);
    return {
      user,
      created: true,
    };
  }

  return {
    user,
    created: false,
  };
};
export { updateUser };
