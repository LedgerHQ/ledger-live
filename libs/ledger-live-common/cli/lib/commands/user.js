"use strict";
exports.__esModule = true;
var rxjs_1 = require("rxjs");
var user_1 = require("@ledgerhq/live-common/lib/user");
exports["default"] = {
    args: [],
    job: function () { return (0, rxjs_1.concat)((0, rxjs_1.of)(JSON.stringify((0, user_1.getUserHashes)()))); }
};
//# sourceMappingURL=user.js.map