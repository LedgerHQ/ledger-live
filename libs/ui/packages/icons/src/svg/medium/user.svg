<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M4.488 19.56V21.36H19.512V19.56C19.488 17.784 19.176 16.608 18.432 15.696C17.448 14.4 15.816 13.68 13.968 13.68H10.032C8.184 13.68 6.6 14.4 5.592 15.696C4.848 16.632 4.512 17.808 4.488 19.56ZM6.528 19.56V18.36C6.528 16.392 7.272 15.72 9.36 15.72H14.664C16.728 15.72 17.472 16.392 17.472 18.36V19.56H6.528ZM7.152 7.36802C7.152 10.032 9.336 12.096 12 12.096C14.664 12.096 16.848 10.032 16.848 7.36802C16.848 4.70401 14.664 2.64001 12 2.64001C9.336 2.64001 7.152 4.70401 7.152 7.36802ZM9.072 7.36802C9.072 5.73601 10.368 4.44001 12 4.44001C13.632 4.44001 14.928 5.73601 14.928 7.36802C14.928 9.00002 13.632 10.296 12 10.296C10.368 10.296 9.072 9.00002 9.072 7.36802Z" fill="black"/>
</svg>
