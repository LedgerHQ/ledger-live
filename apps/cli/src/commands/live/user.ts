import { of, concat } from "rxjs";
import { getUserHashes } from "@ledgerhq/live-common/user";
import { getEnv } from "@ledgerhq/live-env";
export default {
  args: [],
  job: () => concat(of(JSON.stringify(getUserHashes(getEnv("USER_ID"))))),
};
