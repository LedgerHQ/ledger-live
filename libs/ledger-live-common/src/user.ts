import { getEnv } from "@ledgerhq/live-env";
import { sha256 } from "./crypto";

const userHashesPerUserId = (userId: string) => {
  const firmwareSalt = sha256(userId + "|firmwareSalt")
    .toString("hex")
    .slice(0, 6);
  const endpointOverrides100 = sha256(userId + "|endpoint").readUInt16BE(0) % 100;
  return {
    firmwareSalt,
    endpointOverrides100,
  };
};

let cache;
export const getUserHashes = () => {
  const userId = getEnv("USER_ID");

  if (cache && userId === cache.userId) {
    return cache.value;
  }

  cache = {
    userId,
    value: userHashesPerUserId(userId),
  };
  return cache.value;
};
