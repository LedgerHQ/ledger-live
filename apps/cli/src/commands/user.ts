import { of, concat } from "rxjs";
import { getUserHashes } from "@ledgerhq/live-common/user";
export default {
  args: [],
  job: () => concat(of(JSON.stringify(getUserHashes()))),
};
