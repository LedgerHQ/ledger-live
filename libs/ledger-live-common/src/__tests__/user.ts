import { getUserHashes } from "../user";
import { getEnv } from "@ledgerhq/live-env";
test("stable user", () => {
  expect(getUserHashes(getEnv("USER_ID"))).toMatchSnapshot();
});
