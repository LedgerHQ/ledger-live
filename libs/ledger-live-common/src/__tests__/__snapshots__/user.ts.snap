// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stable user 1`] = `
{
  "endpointOverrides100": 20,
  "firmwareSalt": "544d89",
}
`;
